### ISSUE TITLE
[DDP-Dynamo] AotAutograd input mutations due to DDPOptimizer

### ISSUE THREAD

**Issue by wconstab** [pytorch/pytorch#93615]
DDPOptimizer introduces graph-breaks, each of which has 'input' nodes for any live values at the point of the graph-break,  and this increases the odds that we'll step on a pre-existing AotAutograd bug where mutation ops on graph inputs aren't supported and cause compilation to be aborted.  The fallout in the DDPOptimizer case is that a particular subgraph may fail to compile and execute eagerly instead, with a performance impact but no correctness impact.  The fix (from the AotAutograd side) is ongoing.

Here are a couple of repros - (1) only happens with DDPOptimizer graph-breaks, (2) is happening regardless.  Both seem like they would be fixed by AotAutograd improvements to handle input mutation but need to be confirmed.

1)
`python benchmarks/dynamo/torchbench.py --training --accuracy  --backend inductor --ddp --verbose --only resnet50`

this one you have to use —verbose and confirm that you’re not seeing that autograd warning and you’re able to compile the different subgraphs of  the split ddp graph

2)
`python benchmarks/dynamo/torchbench.py --training --accuracy  --ddp --backend inductor --only timm_resnest`

This one fails outright, so you should see an explicit error currently

**Comment by wconstab** [comment]
@bdhirsh i think we can close this now? reopen if not

**Comment by bdhirsh** [comment]
Yep- these should both be fixed
